File: lib/path.c
#include "path.h"

#include <string.h>

void pathunhide(char* path) {
	for (char* chr = path; *chr; chr++) {
		if (*chr == '.' && (chr == path || chr[-1] == '/') && chr[1] != '/')
			*chr = '-';
	}
}

File: path.go
package gitin

func pathunhide(path string) string {
	runes := []rune(path)
	for i, chr := range runes {
		if chr == '.' && (i == 0 || runes[i-1] == '/' && runes[i+1] != '/') {
			runes[i] = '-'
		}
	}
	return string(runes)
}
